foxtrot-core/src/main/java/com/flipkart/foxtrot/core/cache/impl/DistributedCacheFactory.java

```diff
@@ -33,6 +33,7 @@
     public DistributedCacheFactory(HazelcastConnection connection, ObjectMapper mapper) {
         this.connection = connection;
         this.mapper = mapper;
+        DistributedCache.setup(connection);
```

Comment:
Don't do this. setup the cache in the constructor itself. or bring the default setup here and set it up here. Doesnt look right to just put in a static function in the otherwise instantiable class.